tests/testenv/cli_test_env.go

```diff
@@ -106,6 +109,39 @@ func (e *CLITest) RunAndExpectSuccess(t *testing.T, args ...string) []string {
 	return stdout
 }
 
+// TweakFile writes a 0x00 byte at a random point in a file.  Used to simulate file corruption
+func (e *CLITest) TweakFile(t *testing.T, dirn, fglob string) {
+	t.Helper()
+
+	const RwUserGroupOther = 0o666
+	// find a file within the repository to corrupt
+	mch, err := fs.Glob(os.DirFS(dirn), fglob)
+	require.NoError(t, err)
+	require.Greater(t, len(mch), 0)
+	// grab a random file in the directory dirn
+	fn := mch[rand.Intn(len(mch))]
+	f, err := os.OpenFile(path.Join(dirn, fn), os.O_RDWR, os.FileMode(RwUserGroupOther))
+	require.NoError(t, err)
+	// find the length of the file, then seek to a random location
+	l, err := f.Seek(0, io.SeekEnd)
+	require.NoError(t, err)
+	i := rand.Int63n(l)
+	bs := [1]byte{}
+	for {
+		// find a location that isn't already
+		// the value we want to write
+		_, err := f.ReadAt(bs[:], i)
+		require.NoError(t, err)
+		if bs[0] != 0x00 {
+			break
+		}
+		i = rand.Int63n(l)
+	}
```

Comment:
Before we enter an infinite loop, maybe we should check the file we're tweaking actually has any non-zero bytes in it?

Alternatively you could change the mechanism of simulating file corruption to XOR a random byte, which works no matter the content of the file (plus you won't need to loop, cause it'll work for any value)